doc/developers/contributing.rst

```diff
@@ -251,119 +242,138 @@ then submit a "pull request" (PR):
 
        $ git push -u origin my-feature
 
-7. Follow `these
+9. Follow `these
    <https://help.github.com/articles/creating-a-pull-request-from-a-fork>`_
    instructions to create a pull request from your fork. This will send an
    email to the committers. You may want to consider sending an email to the
    mailing list for more visibility.
 
 .. note::
 
-  If you are modifying a Cython module, you have to re-run step 4 after modifications
+  If you are modifying a Cython module, you have to re-run step 5 after modifications
   and before testing them.
 
+It is often helpful to keep your local branch synchronized with the latest
+changes of the main scikit-learn repository::
 
-If any of the above seems like magic to you, then look up the `Git documentation
-<https://git-scm.com/documentation>`_ and the `Git development workflow
-<http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html>`_ on the
-web, or ask a friend or another contributor for help.
+    $ git fetch upstream
+    $ git merge upstream/master
 
-If some conflicts arise between your branch and the ``master`` branch, you need
-to merge ``master``. For that, you first need to fetch the ``upstream``, and
-then merge its ``master`` into your branch::
-
-  $ git fetch upstream
-  $ git merge upstream/master
-
-Subsequently, you need to solve the conflicts. You can refer to the `Git
-documentation related to resolving merge conflict using the command line
+Subsequently, you might need to solve the conflicts. You can refer to the
+`Git documentation related to resolving merge conflict using the command
+line
 <https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/>`_.
 
-.. note::
-
-   In the past, the policy to resolve conflicts was to rebase your branch on
-   ``master``. GitHub interface deals with merging ``master`` better than in
-   the past.
+The `Git documentation <https://git-scm.com/documentation>`_ is an excellent
```

Comment:
For readability, I would put this in a "topic" directive.

Also, I would add a link to http://try.github.io/, which I find really good to learn git + github workflow.